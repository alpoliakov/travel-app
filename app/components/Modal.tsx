import { motion } from 'framer-motion';
import React, { useState } from 'react';
import { useIntl } from 'react-intl';

const postVariants = {
  initial: { scale: 0.96, y: 30, opacity: 0 },
  enter: {
    scale: 1,
    y: 0,
    opacity: 1,
    transition: { duration: 0.5, ease: [0.48, 0.15, 0.25, 0.96] },
  },
  exit: {
    scale: 0.6,
    y: 100,
    opacity: 0,
    transition: { duration: 0.2, ease: [0.48, 0.15, 0.25, 0.96] },
  },
};

export default function Modal({ setModal, propText }) {
  const { formatMessage: f } = useIntl();

  return (
    <div className="w-full h-full flex items-center justify-center fixed block top-0 left-0 bg-gray-400 opacity-80 z-40">
      <motion.div
        className="w-full w-5/6 sm:5/6 md:w-3/4 xl:w-1/2 2xl:w-1/3"
        initial="initial"
        animate="enter"
        exit="exit"
        variants={postVariants}>
        <div className="leading-loose">
          <div className="m-4 relative p-10 bg-gray-800 opacity-99 z-50 rounded shadow-xl">
            <h1 className="text-white text-2xl font-bold">{f({ id: 'hello' })}</h1>
            <span className="text-white text-xl font-semibold">{f({ id: propText })}</span>
            <button
              className="text-white text-xl hover:text-yellow-500 focus:outline-none absolute bottom-3 right-10 block"
              onClick={() => setModal(false)}>
              {f({ id: 'close' })}
            </button>
          </div>
        </div>
      </motion.div>
    </div>
  );
}
