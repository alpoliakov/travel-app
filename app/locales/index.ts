export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    forgetPass: 'Forgot password ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    forgetPass: 'Забыли пароль ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    forgetPass: 'Забули пароль ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
  },
};
