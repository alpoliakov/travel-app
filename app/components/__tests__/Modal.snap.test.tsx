import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { messages } from '../../locales';
import Modal from '../Modal';

const mockFn = jest.fn();

const REMAINDER = 'reminder';

it('Modal renders correctly', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="ru" messages={messages['ru']}>
        <Modal setShowModal={mockFn} propText={REMAINDER} />)
      </IntlProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
