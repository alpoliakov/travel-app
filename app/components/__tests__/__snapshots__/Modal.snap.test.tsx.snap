// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
Array [
  <div
    className="w-full h-full flex items-center justify-center fixed block top-0 left-0 bg-gray-400 opacity-80 z-40"
  >
    <div
      className="w-full w-5/6 sm:5/6 md:w-3/4 xl:w-1/2 2xl:w-1/3"
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(30px) scale(0.96) translateZ(0)",
        }
      }
    >
      <div
        className="leading-loose"
      >
        <div
          className="m-4 relative p-10 bg-gray-800 opacity-99 z-50 rounded shadow-xl"
        >
          <h1
            className="text-white text-2xl font-bold"
          >
            Приветствуем!
          </h1>
          <span
            className="text-white text-xl font-semibold"
          >
            Для просмотра необходимо авторизоваться или зарегистрироваться!
          </span>
          <button
            className="text-white text-xl hover:text-yellow-500 focus:outline-none absolute bottom-3 right-10 block"
            onClick={[Function]}
          >
            Закрыть
          </button>
        </div>
      </div>
    </div>
  </div>,
  ")",
]
`;
